### PR TITLE
Less threads, less DB handles

### PR DESCRIPTION
**Pull request by tylerkaraszewski**
### Fixed Issues
Fixes GH_LINK

### Tests

_________
**Internal Testing Reminder:** when changing bedrock, please compile auth against your new changes

### CODE REVIEW

File: BedrockServer.cpp [hunk 1]
```diff
@@ -1857,6 +1870,28 @@ void BedrockServer::_control(unique_ptr<BedrockCommand>& command) {
             __quiesceThread = nullptr;
             response.methodLine = "200 Unblocked";
         }
+    } else if (SIEquals(command->request.methodLine, "SetMaxSocketThreads")) {
+        size_t newMax = command->request.calcU64("threads");
```
Comment (flodnv) [inline BedrockServer.cpp:1874]:
```suggestion
        size_t newMax = command->request.calcU64("socketThreadsCount");
```

File: BedrockServer.cpp [hunk 2]
```diff
@@ -1857,6 +1870,28 @@ void BedrockServer::_control(unique_ptr<BedrockCommand>& command) {
...
+        if (newMax) {
+            SINFO("Setting _maxSocketThreads to " << newMax << " from " << _maxSocketThreads);
+            _maxSocketThreads = newMax;
+        } else {
+            response.methodLine = "401 Don't Use Zero";
+        }
+    } else if (SIEquals(command->request.methodLine, "SetMaxDBPoolSize")) {
```
Comment (flodnv) [inline BedrockServer.cpp:1881]:
```suggestion
    } else if (SIEquals(command->request.methodLine, "SetMaxDBHandles")) {
```

File: BedrockServer.cpp [hunk 3]
```diff
@@ -1857,6 +1870,28 @@ void BedrockServer::_control(unique_ptr<BedrockCommand>& command) {
...
+        shared_ptr<SQLitePool> dbPoolHandle = _dbPool;
+        if (dbPoolHandle) {
+            size_t newMax = command->request.calcU64("handles");
```
Comment (flodnv) [inline BedrockServer.cpp:1884]:
```suggestion
            size_t newMax = command->request.calcU64("handlesCount");
```

File: BedrockServer.cpp [hunk 4]
```diff
@@ -1982,6 +2017,11 @@ void BedrockServer::_acceptSockets() {
     // Try block because we sometimes catch `std::system_error` from in here (likely from the thread code) and we're
     // trying to diagnose exactly what's happening.
     try {
+        if (_outstandingSocketThreads >= _maxSocketThreads) {
+            SINFO("Not accepting any new socket threads as we already have " << _outstandingSocketThreads << " of " << _maxSocketThreads);
```
Comment (flodnv) [inline BedrockServer.cpp:2021]: Can we have this log line for db handles? Or we already have it?

File: BedrockServer.h [hunk 5]
```diff
@@ -470,4 +470,7 @@ class BedrockServer : public SQLiteServer {
     // This is just here to allow `poll` in main.cpp to get interrupted when the server shuts down.
     // to wait up to a full second for them.
     SSynchronizedQueue<bool> _notifyDone;
+
+    atomic<size_t> _maxSocketThreads{500};
```
Comment (flodnv) [inline BedrockServer.h:474]: Can we make this higher, like a middle ground between this and the previous 25k one?

File: BedrockServer.cpp (hunk 4 above)
Comment (tylerkaraszewski) [reply BedrockServer.cpp:2021]: We already have a similar log line.

File: BedrockServer.h (hunk 5 above)
Comment (tylerkaraszewski) [reply BedrockServer.h:474]: There was no previous maximum here. My reasoning with this is that things are already looking broken when this number is in the low 1000s. We could make it 1000?

**Review by flodnv — APPROVED** [review]

**Pull request merged**